core/src/main/java/fr/sncf/osrd/cli/StandaloneSimulationCommand.java

```
@@ -0,0 +1,133 @@
+import fr.sncf.osrd.railjson.schema.schedule.*;
+import fr.sncf.osrd.reporting.warnings.WarningRecorderImpl;
+import fr.sncf.osrd.standalone_sim.StandaloneSim;
+import fr.sncf.osrd.standalone_sim.result.StandaloneSimResult;
+import fr.sncf.osrd.standalone_sim.result.StandaloneSimResultAllTrains;
+import fr.sncf.osrd.train.RollingStock;
+import fr.sncf.osrd.train.StandaloneTrainSchedule;
+import java.io.FileWriter;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+
+public class StandaloneSimulationCommand implements CliCommand {
+
+    @Parameter(
+            names = { "infra_path" },
+            description = "Path to the infra railjson file to load",
+            required = true
+    )
+    private String infraFilePath;
+
+    @Parameter(
+            names = { "sim_path" },
+            description = "Path to the sim railjson file to load",
+            required = true
+    )
+    private String simFilePath;
+
+    @Parameter(
+            names = { "res_path" },
+            description = "Path to the result file to save",
+            required = true
+    )
+    private String resultFilePath;
+
+    @Override
+    public int run() {
+
+        try {
+            // Parse the entry files from json
+            var rjsInfra = RJSParser.parseRailJSONFromFile(infraFilePath);
+            var warningRecorder = new WarningRecorderImpl(true);
+            final var signalingInfra = SignalingInfraBuilder.fromRJSInfra(
+                    rjsInfra,
+                    Set.of(new BAL3(warningRecorder)),
+                    warningRecorder);
+            warningRecorder.report();
+
+            var simFile = Files.readString(Path.of(simFilePath));
+            var sim = RJSSimulation.adapter.fromJson(simFile);
+
+            // Parse rolling stocks
+            var rollingStocks = new HashMap<String, RollingStock>();
+            assert sim != null;
```

You can check that `sim` is not null directly after serializing it.
If it's null you should throw an exception instead of an `AssertError`.